target_chains/sui/contracts/sources/pyth.move

```diff
@@ -953,52 +926,158 @@ module pyth::pyth_tests{
             ctx(&mut scenario)
         );
 
-        // Affirm that 1 object, which correspond to the 1 new price info object
-        // containing the price feeds were created and shared.
-        let effects = test_scenario::next_tx(&mut scenario, DEPLOYER);
-        let shared_ids = test_scenario::shared(&effects);
-        let created_ids = test_scenario::created(&effects);
-        assert!(vector::length<ID>(&shared_ids)==1, 0);
-        assert!(vector::length<ID>(&created_ids)==1, 0);
+        // clean up test scenario
+        test_scenario::next_tx(&mut scenario, DEPLOYER);
+        coin::burn_for_testing<SUI>(coins);
 
-        let price_info_object_1 = take_shared<PriceInfoObject>(&scenario);
+        cleanup_worm_state_pyth_state_and_clock(worm_state, pyth_state, clock);
+        test_scenario::end(scenario);
+    }
 
-        // Create authenticated price infos
-        verified_vaa = get_verified_vaa_from_accumulator_message(&worm_state, TEST_ACCUMULATOR_SINGLE_FEED, &clock);
-        let auth_price_infos = pyth::create_authenticated_price_infos_using_accumulator(
-            &pyth_state,
-            TEST_ACCUMULATOR_SINGLE_FEED,
+    #[test_only]
+    const TEST_ACCUMULATOR_INVALID_PROOF_1: vector<u8> = x"504e41550100000000a001000000000100110db9cd8325ccfab0dae92eeb9ea70a1faba5c5e96dc21ff46a8ddc560afc9a60df096b8ff21172804692bbdc958153e838437d8b474cbf45f0dc2a8acae831000000000000000000000171f8dcb863d176e2c420ad6610cf687359612b6fb392e0642b0ca6b1f186aa3b0000000000000000004155575600000000000000000000000000a8bea2b5f12f3177ff9b3929d77c3476ab2d32c602005500b10e2d527612073b26eecdfd717e6a320cf44b4afac2b0732d9fcbe2b7fa0cf6fa75cd3aa3bb5ace5e2516446f71f85be36bd19bb0703f3154bb3db07be59f3f377c0d3f44661d9a8736c68884c8169e8b636ee3043202397384073120dce9e5d0efe24b44b4a0d62da8a1180177cf30b2c0bebbb1adfe8f7985d051d205a01e2504d9f0c06e7e7cb0cf24116098ca202ac5f6ade2e8f5a12ec006b16d46be1f0228b94d950055006e1540171b6c0c960b71a7020d9f60077f6af931a8bbf590da0223dacf75c7af5f958f4883f9d2a8b5b1008d1fa01db95cf4a8c7423a695e81ad1eb504f8554c3620c3fd40b40f7d581ac802e2de5cb82a9ae672043202397384073120dce9e5d0efe24b44b4a0d62da8a1180177cf30b2c0bebbb1adfe8f7985d051d205a01e2504d9f0c06e7e7cb0cf24116098ca202ac5f6ade2e8f5a12ec006b16d46be1f0228b94d95";
+
+     #[test]
+    #[expected_failure(abort_code = pyth::accumulator::E_INVALID_PROOF)]
+    fun test_accumulator_invalid_proof() {
```

Comment:
can these be cut down even more in terms of code duplication? it seems like the key variables are the setup and the message bytes, and then there's quite a bit of additional boilerplate.

Reply:
Yes, will do another compression in the end in a separate PR